### PR TITLE
Branch 154631802

### PR DESCRIPTION
**Pull request by caisq**

### CODE REVIEW

**Comment by caisq** [comment]
Looks like this test failure is related to PR #8461. Any thoughts, @karandesai-96 ?

```
======================================================================
FAIL: testConv3DTransposeBiasRegularizer (__main__.Conv3DTransposeTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/bazel_pip/tensorflow/python/layers_convolutional_test.runfiles/org_tensorflow/bazel_pip/tensorflow/python/layers/convolutional_test.py", line 763, in testConv3DTransposeBiasRegularizer
    self.assertListEqual(layer.losses, loss_keys)
AssertionError: Lists differ: [] != [<tf.Tensor 'conv3d_transpose/...

Second list contains 1 additional elements.
First extra element 0:
Tensor("conv3d_transpose/bias/Regularizer/mul:0", shape=(), dtype=float32)

- []
+ [<tf.Tensor 'conv3d_transpose/bias/Regularizer/mul:0' shape=() dtype=float32>]

======================================================================
FAIL: testConv3DTransposeKernelRegularizer (__main__.Conv3DTransposeTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/var/lib/jenkins/workspace/tensorflow-pull-requests-cpu/bazel-ci_build-cache/.cache/bazel/_bazel_jenkins/eab0d61a99b6696edb3d2aff87b585e8/execroot/workspace/bazel-out/local-opt/bin/bazel_pip/tensorflow/python/layers_convolutional_test.runfiles/org_tensorflow/bazel_pip/tensorflow/python/layers/convolutional_test.py", line 753, in testConv3DTransposeKernelRegularizer
    self.assertListEqual(layer.losses, loss_keys)
AssertionError: Lists differ: [] != [<tf.Tensor 'conv3d_transpose/...

Second list contains 1 additional elements.
First extra element 0:
Tensor("conv3d_transpose/kernel/Regularizer/mul:0", shape=(), dtype=float32)

- []
+ [<tf.Tensor 'conv3d_transpose/kernel/Regularizer/mul:0' shape=() dtype=float32>]

----------------------------------------------------------------------
Ran 73 tests in 3.422s

FAILED (failures=2)
================================================================================
```

**Comment by karandesai-96** [comment]
Hi @caisq, this is strange. I pulled your branch locally and tested it. I could not reproduce the error. Bazel tests passed successfully.

**Comment by caisq** [comment]
@karandesai-96 Let me run it again and see if it is reproducible or flaky. This may have to do with some of the layers refactoring in this push.

cc @fchollet

**Comment by caisq** [comment]
@tensorflow-jenkins test this please

**Comment by fchollet** [comment]
Note that after the refactoring, you should create layer variables via the `add_variable` method (see other layers for examples). This takes care of keeping track of the weights and losses.

**Comment by caisq** [comment]
@fchollet @karandesai-96 Since it is just a two-line fix, can you send me the diff so that I can add a commit to this push to fix it?

**Comment by fchollet** [comment]
In `Conv3DTranspose`, replace

```python
 +    self.kernel = vs.get_variable('kernel',
 +                                  shape=kernel_shape,
 +                                  initializer=self.kernel_initializer,
 +                                  regularizer=self.kernel_regularizer,
 +                                  trainable=True,
 +                                  dtype=self.dtype)
 +    if self.use_bias:
 +      self.bias = vs.get_variable('bias',
 +                                  shape=(self.filters,),
 +                                  initializer=self.bias_initializer,
 +                                  regularizer=self.bias_regularizer,
 +                                  trainable=True,
 +                                  dtype=self.dtype)
```

with

```python
 +    self.kernel = self.add_variable(name='kernel',
 +                                  shape=kernel_shape,
 +                                  initializer=self.kernel_initializer,
 +                                  regularizer=self.kernel_regularizer,
 +                                  trainable=True,
 +                                  dtype=self.dtype)
 +    if self.use_bias:
 +      self.bias = self.add_variable(name='bias',
 +                                  shape=(self.filters,),
 +                                  initializer=self.bias_initializer,
 +                                  regularizer=self.bias_regularizer,
 +                                  trainable=True,
 +                                  dtype=self.dtype)
```

**Comment by caisq** [comment]
@fchollet Thanks. Done.

File: tensorflow/python/layers/convolutional.py [hunk 1]
```diff
@@ -1282,14 +1282,14 @@ def build(self, input_shape):
     input_dim = input_shape[channel_axis]
     kernel_shape = self.kernel_size + (self.filters, input_dim)
 
-    self.kernel = vs.get_variable('kernel',
+    self.kernel = vs.add_variable('kernel',
```
Comment (fchollet) [inline tensorflow/python/layers/convolutional.py]: It's `self.add_variable`. It's a layer method.

Comment (caisq) [inline tensorflow/python/layers/convolutional.py]: oops. sorry. done.

**Comment by caisq** [comment]
@tensorflow-jenkins test this please

**Review by vrv — APPROVED** [review]

**Pull request merged**